services/movie.py

```diff
@@ -0,0 +1,42 @@
+from django.db.models import QuerySet, Q
+
+from db.models import Movie, Genre, Actor
+
+
+def get_movies(
+        genres_ids: list[int] = None,
+        actors_ids: list[int] = None
+) -> QuerySet | Movie:
+    queryset = Movie.objects.all()
+    if not genres_ids and not actors_ids:
+        return queryset
+    if genres_ids and actors_ids:
+        return queryset.filter(
+            Q(genres__id__in=genres_ids) & Q(actors__id__in=actors_ids)
+        ).distinct()
+    if genres_ids:
+        return queryset.filter(genres__id__in=genres_ids).distinct()
+    if actors_ids:
+        return queryset.filter(actors__id__in=actors_ids).distinct()
+    return queryset
```

Comment:
you can simplify this code. isn't it easier to check if some parameter is passed and if yes, then filter queryset but don't return it and go to the next if statement?

Reply:
also you can do distinct at the end in order do decrease sql queries to db